File: desktop/src/main/scala/synth/synth.rb
require 'java'

$Seer = Java::com.fishuyo
$Key = $Seer.io.Keyboard
$Trackpad = $Seer.io.Trackpad
$Vec3 = $Seer.maths.Vec3
$Quat = $Seer.maths.Quat
$Camera = $Seer.graphics.Camera
$Audio = $Seer.audio.Audio
$Sine = $Seer.audio.Sine

$Main = $Seer.examples.synth.Main
sines = $Main.sines
osc = $Main.osc
lfo = $Main.lfo

$Key.clear()
$Key.use()
$Key.bind("n", lambda{
	#gen = ($Sine.new(1.0,100.0)).->($Sine.new(440,1))
	#$Audio.push( gen )
})

dx = 0.0
dy = 0.0
$Trackpad.clear()
$Trackpad.connect()
$Trackpad.bind( lambda{ |i,f| 
	for s in 0...i
		break if s >= sines.length()
		#puts sines[s].frequency
		sines[s].f(f[5+2*s] * 2000.0)
		sines[s].a(f[5+2*s+1] * 1.0)
	end

	if i == 3

	elsif i == 2	
		dx += f[2] * 0.1	
		dy += f[3] * -0.1
		lfo.f(dx*0.0)
		lfo.a(dy*0.0)
		osc.a(0.0)
		#osc.f(f[1]*0.0)
	else
	end
})



def step(dt)
	#puts $Seer.audio.Audio.dt
end

File: experiments/synth/synth.scala
package com.fishuyo.seer
package examples.synth
import maths._
import graphics._
import trees._
import io._
import dynamic._
import audio._

import scala.concurrent.duration._
import akka.util.Timeout
import akka.pattern.ask

import scala.collection.mutable.ListBuffer

object Main extends App with Animatable {

  DesktopApp.loadLibs()
  Scene.push(this)

  var sines = ListBuffer[Osc]()
  for( i<-(0 until 10)) sines += new Sine(0.f)
  sines.foreach( s => Audio.push( s ))

  var display = Array(new AudioDisplay(512), new AudioDisplay(512))
  display(0).pose.pos.set(-1,0,0)

  var osc = new Tri(41.f)
  var lfo = new Saw(1.f)

  Audio.push( (lfo + osc.f ) -> osc  ) //* ((1.f-lfo)*.5f) )

  val live = new Ruby("synth.rb")

  DesktopApp()

  override def step(dt:Float){

    // implicit val timeout = Timeout(5 seconds)

    live.step(dt)

    //val f = Audio.main ? OutBuffer
    //f.zipWithIndex.foreach( (i,a) => display(i).setSamples(a) )

    display(0).setSamples(Audio.out(0))
    display(1).setSamples(Audio.out(1))
  } 
  override def draw(){
    Shader.lightingMix = 0
  	display.foreach( _.draw())
  }
}



